Fixes #26047 - Add check for puppet in cvv export.

chris1984 opened the pull request:
This PR adds a check to export, to check and see if a cv version contains puppet modules and give the user a friendly error instead of having `tar` complain with something that is confusing. 

I also re-factored the export checks into a method in the helper file to clean up the code within the content_view_version file to allow for more readability and flexibility for adding more checks etc that come up over time.

```
CV with only Puppet modules

[vagrant@centos7-katello-devel ~]$ hammer content-view version export --id 2 --export-dir /tmp
Could not export the content view:
  Error: The Content View 'puppet-view' contains Puppet modules, this is not supported at this time. Please remove the modules, publish a new version and try the export again.

CV with Puppet modules and yum repos

[vagrant@centos7-katello-devel ~]$ hammer content-view version export --id 4 --export-dir /tmp
Could not export the content view:
  Error: The Content View 'mixed-zoo' contains Puppet modules, this is not supported at this time. Please remove the modules, publish a new version and try the export again.

Verified checks still work after moving them to the helper file

[vagrant@centos7-katello-devel ~]$ hammer content-view version export --id 6 --export-dir /tmp
Could not export the content view:
  Error: All exported repositories must be set to an immediate download policy and re-synced.
  The following repositories need action:
    background
[vagrant@centos7-katello-devel ~]$ hammer content-view version export --id 7 --export-dir /tmp
Could not export the content view:
  Error: The Repository 'filerepo' is a non-yum repository. Only Yum is supported at this time. Please remove the repository from the Content View, republish and try the export again.
```

akofink commented on lib/hammer_cli_katello/content_view_version.rb:
```diff
@@ -349,32 +348,6 @@ def create_tar(cv, cvv, repositories, json)
-        end
-      end
-
-      def check_repo_download_policy(repositories)
```
I personally like these methods staying separate, really just because it helps to document what they do. I'm perfectly fine with you moving them to the helper and creating a new method to call them. The new logic you've added could be in a third method, named by what it does.

chris1984 commented:
@akofink updated and addressed your comments. Confirmed tests still work:

```
[vagrant@centos7-katello-devel ~]$ hammer content-view version list
---|-------------------------------|---------|-----------------------
ID | NAME                          | VERSION | LIFECYCLE ENVIRONMENTS
---|-------------------------------|---------|-----------------------
7  | file-view 1.0                 | 1.0     | Library               
6  | background 1.0                | 1.0     | Library               
5  | on-demand 1.0                 | 1.0     | Library               
4  | mixed-zoo 1.0                 | 1.0     | Library               
3  | zoo 1.0                       | 1.0     | Library               
2  | puppet-view 1.0               | 1.0     | Library               
1  | Default Organization View 1.0 | 1.0     | Library               
---|-------------------------------|---------|-----------------------
[vagrant@centos7-katello-devel ~]$ hammer content-view version export --id 2 --export-dir /tmp
Could not export the content view:
  Error: The Content View 'puppet-view' contains Puppet modules, this is not supported at this time. Please remove the modules, publish a new version and try the export again.
[vagrant@centos7-katello-devel ~]$ hammer content-view version export --id 4 --export-dir /tmp
Could not export the content view:
  Error: The Content View 'mixed-zoo' contains Puppet modules, this is not supported at this time. Please remove the modules, publish a new version and try the export again.
[vagrant@centos7-katello-devel ~]$ hammer content-view version export --id 5 --export-dir /tmp
Could not export the content view:
  Error: All exported repositories must be set to an immediate download policy and re-synced.
  The following repositories need action:
    medium
[vagrant@centos7-katello-devel ~]$ hammer content-view version export --id 7 --export-dir /tmp
Could not export the content view:
  Error: The Repository 'filerepo' is a non-yum repository. Only Yum is supported at this time. Please remove the repository from the Content View, republish and try the export again.
```

chris1984 commented:
@ehelms care to review when you get a moment?
